llvm/lib/Transforms/Scalar/MemCpyOptimizer.cpp

```diff
@@ -1681,6 +1687,12 @@ bool MemCpyOptPass::performStackMoveOptzn(Instruction *Load, Instruction *Store,
   for (Instruction *I : NoAliasInstrs)
     I->setMetadata(LLVMContext::MD_noalias, nullptr);
 
+  // Remove !tbaa and !tbaa_struct from the metadata, since they are invalid.
+  for (Instruction *I : OptimizedAllocaInstUsers) {
```

Comment:
Can we combine NoAliasInstrs and OptimizedAllocaInstUsers? The purpose is basically the same, I don't think we need separate sets for different metadata.